File: src/main/java/ch/obermuhlner/astro/image/color/ColorModel.java
package ch.obermuhlner.astro.image.color;

public interface ColorModel {

  ColorModelRGB RGB = new ColorModelRGB();
  ColorModelHSV HSV = new ColorModelHSV();

  class ColorModelRGB implements ColorModel {
    public static final int R = 0;
    public static final int G = 1;
    public static final int B = 2;
  }

  class ColorModelHSV implements ColorModel {
    public static final int H = 0;
    public static final int S = 1;
    public static final int V = 2;
  }
}

File: src/main/java/ch/obermuhlner/astro/image/color/ColorModel.kt
package ch.obermuhlner.astro.image.color

interface ColorModel {
    object RGB : ColorModel {
        const val R = 0
        const val G = 1
        const val B = 2
    }

    object HSV : ColorModel {
        const val H = 0
        const val S = 1
        const val V = 2
    }
}